### PR TITLE
[BEAM-2125] update JavaDoc of BoundedWindow

### PR DESCRIPTION
**Pull request by XuMingmin**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

**Comment by XuMingmin** [comment]
R: @kennknowles

**Comment by XuMingmin** [comment]
Retest this please

File: sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/BoundedWindow.java [hunk 1]
```diff
@@ -21,11 +21,8 @@
 import org.joda.time.Instant;
 
 /**
- * A {@code BoundedWindow} represents a finite grouping of elements, with an
- * upper bound (larger timestamps represent more recent data) on the timestamps
- * of elements that can be placed in the window. This finiteness means that for
- * every window, at some point in time, all data for that window will have
- * arrived and can be processed together.
+ * A {@code BoundedWindow} represents window information assigned to data elements.
+ * At some point in time, all data with the same window will arrive and can be processed together.
```
Comment (kennknowles) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/BoundedWindow.java]: I think if you remove the discussion of "upper bound" I think replace it with something, since having a `maxTimestamp()` is the main feature of windows.

Basically the `maxTimestamp()` is the default event time where the window is expected to be complete, and it can be garbage collected unless allowed lateness is set.

Comment (XuMingmin) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/BoundedWindow.java]: updated to include info of `maxTimestamp()`, please share any suggestion.

File: sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/BoundedWindow.java [hunk 2]
```diff
@@ -21,11 +21,10 @@
...
 
 /**
- * A {@code BoundedWindow} represents a finite grouping of elements, with an
- * upper bound (larger timestamps represent more recent data) on the timestamps
- * of elements that can be placed in the window. This finiteness means that for
- * every window, at some point in time, all data for that window will have
- * arrived and can be processed together.
+ * A {@code BoundedWindow} represents window information assigned to data elements.
+ * It has one method {@link #maxTimestamp()} to define an upper bound(included).
+ * Elements within the timestamp range are placed in this window.
```
Comment (kennknowles) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/BoundedWindow.java]: Since you are editing this, I will give you all of my suggestions, even expanding it a lot.

A window technically does not have a lower bound, which is worth noting.

And the only reason an element timestamp has to be below the upper bound is because the watermark is based on element timestamps and the watermark also controls discarding data and expiring the window.

File: sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/BoundedWindow.java [hunk 3]
```diff
@@ -21,11 +21,10 @@
 import org.joda.time.Instant;
 
 /**
- * A {@code BoundedWindow} represents a finite grouping of elements, with an
- * upper bound (larger timestamps represent more recent data) on the timestamps
- * of elements that can be placed in the window. This finiteness means that for
- * every window, at some point in time, all data for that window will have
- * arrived and can be processed together.
+ * A {@code BoundedWindow} represents window information assigned to data elements.
```
Comment (kennknowles) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/BoundedWindow.java]: I think the subject line for the class should be a bit shorter and then newline and `<p>`

File: sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/BoundedWindow.java (hunk 2 above)
Comment (XuMingmin) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/BoundedWindow.java]: good points! Let me rephrase it.
